run.py

```diff
@@ -95,130 +100,161 @@
-
-    bounce_count += 1
-    point_detector.bounce_detected(ball_in_bounds)
-    display_bounce = 0
-    #point_detector.side_of_court = side_of_court
-  
-  # if the ball has been out of bounds or not detected for a long time, a point was probably scored 
-  if point_detector.frames_out_of_bounds > 60 and not point_detector.point_was_scored and not first_bounce:
-    point_detector.reset()
-    point_detector.points += 1
-    commentator.set_commentary('point')
-    point_detector.point_was_scored = True
+    point_detector.point_was_scored = False
+    first_hit = True
+
+  # double check for false positive replay
+  if game_status == 2:
+    replay_detector.replay_frames +=1
+    isreplay = replay_detector.confirm_if_replay(highest_conf_objects, point_detector.points)
+    if isreplay:
+      replay_detector.expected_points += 1
+      commentator.set_commentary('replay')
+  else:
+    replay_detector.replay_frames = 0
+
+  # only do this if the game status is in the middle of a rally
+  if game_status == 1:
+
+    # only check for hits until the first hit is detected for now
+    if first_hit:
+      frame_num = math.floor(count * audio_file.fps / video_file.fps)
+      audio_reader.seek(frame_num)
+      audio = audio_utils.get_audio(audio_reader)
+      hit_detected = audio_utils.predict(audio, audio_model)[0] > 0.5
+      if hit_detected:
+          commentator.set_commentary('serve')
+          first_hit = False
+          #display_hit = 0
+
+    # get the status of the current ball data
+    #curr_ball_data, bounce_detected = detect_bounces(boxes,first_ball_data, prev_ball_data, count, event_det_model)
+    curr_ball_data, bounce_detected = detect_bounces(boxes,first_ball_data, prev_ball_data)
+
+    if len(prev_ball_data) > 0:
+      result = get_side_of_court(det, prev_ball_data)
+      if result != None:
+        point_detector.side_of_court = result
+
+    # check if the ball has switched sides
+    if point_detector.side_of_court * point_detector.prev_side_of_court < 0:
+      # reset last bounce, since a double bounce can't happen if ball switches sides
+      point_detector.side_of_last_bounce = 0
+
+    # draw the last 3 positions of the ball
+    if len(curr_ball_data) > 0 and len(prev_ball_data) > 0 and len(first_ball_data) > 0:
+      prev = get_center(prev_ball_data)
+      first = get_center(first_ball_data)
+      curr = get_center(curr_ball_data)
+
+      cv2.circle(image, prev, 3, (255, 255, 0), 1)
+      cv2.circle(image, first, 3, (0, 0, 255), 1)
+      cv2.circle(image, curr, 3, (0, 255, 0), 1)
+
+    # check if the ball is in bounds
+    image, boundaries = get_court_boundary(det, image, show_data=True)
+    image, ball_in_bounds = checkIfBallInBounds(prev_det, image, boundaries, show_data=False)
+
+    # keep track of how many frames in a row the ball is out of bounds
+    #point_detector.frames_out_of_bounds = point_detector.frames_out_of_bounds + 1 if not ball_in_bounds else 0
```

Comment:
Do we need this?

Reply:
I'm okay with the current implementation. If you think there's a reason you'd want to switch back, leave it, otherwise delete this.

Reply:
Ok i'll just leave it for now, and remove in a later PR